src/librustc_trait_selection/traits/fulfill.rs

```diff
@@ -458,16 +459,17 @@ impl<'a, 'b, 'tcx> ObligationProcessor for FulfillProcessor<'a, 'b, 'tcx> {
                 }
 
                 ty::PredicateAtom::ConstEvaluatable(def_id, substs) => {
-                    match self.selcx.infcx().const_eval_resolve(
-                        obligation.param_env,
+                    const_evaluatable::is_const_evaluatable(
+                        self.selcx.infcx(),
                         def_id,
                         substs,
-                        None,
-                        Some(obligation.cause.span),
-                    ) {
-                        Ok(_) => ProcessResult::Changed(vec![]),
-                        Err(err) => ProcessResult::Error(CodeSelectionError(ConstEvalFailure(err))),
-                    }
+                        obligation.param_env,
+                        obligation.cause.span,
+                    )
+                    .map_or_else(
+                        |e| ProcessResult::Error(CodeSelectionError(ConstEvalFailure(e))),
+                        |()| ProcessResult::Changed(vec![]),
```

Comment:
idk, i like matches for these, but if you feel strongly about it, keep your change, if not, let's keep the status quo.